src/lib/util/cdrom.cpp

```diff
@@ -2125,89 +2127,170 @@ std::error_condition cdrom_file::parse_gdi(std::string_view tocfname, toc &outto
 	char linebuffer[512];
 	memset(linebuffer, 0, sizeof(linebuffer));
 
-	fgets(linebuffer,511,infile);
-	numtracks=atoi(linebuffer);
+	if (!fgets(linebuffer,511,infile))
+	{
+		osd_printf_error("GDI doesn't have track count (blank file?)\n");
+		return chd_file::error::INVALID_DATA;
+	}
+
+	i = 0;
+	TOKENIZE
+
+	const int numtracks = atoi(token);
 
-	for(i=0;i<numtracks;++i)
+	if (numtracks > 0 && numtracks - 1 > std::size(outinfo.track))
 	{
-		char *tok;
-		int trknum;
-		int trksize,trktype;
-		int sz;
+		osd_printf_error("GDI expects too many tracks. Expected %d tracks but only up to %zu tracks allowed\n", numtracks, std::size(outinfo.track) + 1);
```

Comment:
You don’t actually need the size qualifiers with the `osd_printf_*` functions.  Using them will cause whatever type is passed in to be cast to the specified size, but omitting them just uses it as-is.